pkg/destroy/openstack/openstack.go

```diff
@@ -320,20 +320,45 @@ func deleteServerGroups(opts *clientconfig.ClientOpts, filter Filter, logger log
 	return numberDeleted == numberToDelete, nil
 }
 
-func deletePorts(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
-	logger.Debug("Deleting openstack ports")
-	defer logger.Debugf("Exiting deleting openstack ports")
+func deletePortsByNetwork(opts *clientconfig.ClientOpts, networkID string, logger logrus.FieldLogger) (bool, error) {
 
-	conn, err := clientconfig.NewServiceClient("network", opts)
+	listOpts := ports.ListOpts{
+		NetworkID: networkID,
+	}
+
+	result, err := deletePorts(opts, listOpts, logger)
 	if err != nil {
 		logger.Error(err)
 		return false, nil
 	}
+	return result, err
+}
+
+func deletePortsByFilter(opts *clientconfig.ClientOpts, filter Filter, logger logrus.FieldLogger) (bool, error) {
+
 	tags := filterTags(filter)
 	listOpts := ports.ListOpts{
 		TagsAny: strings.Join(tags, ","),
```

Comment:
Wait, why is that `TagsAny`? Not a question for this patch, just curious.

Reply:
At least one of the tags must be present. Let's just say it's future proof if we start filtering with more than just the cluster ID :man_shrugging: